milli/src/update/delete_documents.rs

```diff
@@ -100,6 +103,17 @@ impl<'t, 'u, 'i> DeleteDocuments<'t, 'u, 'i> {
             let key = BEU32::new(docid);
             let mut iter = documents.range_mut(self.wtxn, &(key..=key))?;
             if let Some((_key, obkv)) = iter.next().transpose()? {
+                for (field_id, _) in obkv.iter() {
+                    let field_name = fields_ids_map.name(field_id).unwrap();
+                    if let Entry::Occupied(mut entry) = fields_distribution.entry(field_name.to_string()) {
+                        if entry.get() == &1 {
+                            entry.remove();
+                        } else {
+                            *entry.get_mut() -= 1;
+                        }
+                    }
+                }
```

Comment:
I am sure that we could implement a better way to do this by:
 - avoiding retrieving the field names each time we need to access a field in the `fields_distribution` map.
 - creating a HashMap that as the fields ids as keys and increment the count for each deleted document.
 - doing the fields ids conversion to fields names at the end of the loop.
 - doing the clean up after the loop, after the conversion.

This way we reduce the amount of code we add into the main deletion loop (i.e. `or_default() += 1` instead of a match) and the number of lookups in the `fields_ids_map` to fetch the names.

What do you think?

Reply:
Good idea, agree. Fixed, check it out.